tests/data_access.rs

```diff
@@ -28,7 +28,7 @@ struct Loaded {
 }
 
 const TRACKABLE_MAX: u64 = 3600 * 1000 * 1000;
-const SIGFIG: u32 = 3;
+const SIGFIG: u8 = 3;
```

Comment:
Hah, yes, Criterion unfortunately has some way to go in that department (and in some other aspects too). `benchmarks/busy.rs` is not really a benchmark per se, it just records lots and lots of values in a tight loop, which is useful for performance profiling. `benchmarks/all.rs` is the one that actually does profiling. It requires a little bit of juggling because `crates.io` does not allow `git` dependencies

```console
$ git diff Cargo.toml
diff --git a/Cargo.toml b/Cargo.toml
index e16fa3d..118d2f5 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -20,12 +20,12 @@ license = "MIT/Apache-2.0"
 travis-ci = { repository = "jonhoo/hdrsample" }

 [features]
-#benchmark = ["criterion"]
-benchmark = [] # for crates.io publication
+benchmark = ["criterion"]
+#benchmark = [] # for crates.io publication

 [dependencies]
 num = "0.1"
-#criterion = { git = "https://github.com/japaric/criterion.rs.git", optional = true }
+criterion = { git = "https://github.com/japaric/criterion.rs.git", optional = true }

 [lib]
 path = "src/lib.rs"
$ rustup run nightly cargo run --bin perf --release --features benchmark
```

Criterion also does performance regression testing, so it might report a "failure" in some cases. Just ignore that. What is interesting is the performance numbers it spits out. On my machine, I get

```
Benchmarking raw_record_with_interval_u64
> Warming up for 2.0000 s
> Collecting 7000 samples in estimated 5.0201 s
> Found 211 outliers among 6999 measurements (3.01%)
  > 117 (1.67%) high mild
  > 94 (1.34%) high severe
> Performing linear regression
  >  slope [11.496 ns 11.514 ns]
  >    R^2  0.1003064 0.1002968
> Estimating the statistics of the sample
  >   mean [11.566 ns 11.668 ns]
  > median [11.590 ns 11.590 ns]
  >    MAD [12.385 ps 17.341 ps]
  >     SD [558.04 ps 3.9870 ns]

Benchmarking raw_record_u64
> Warming up for 1.0000 s
> Collecting 7000 samples in estimated 5.1764 s
> Found 263 outliers among 6999 measurements (3.76%)
  > 97 (1.39%) high mild
  > 166 (2.37%) high severe
> Performing linear regression
  >  slope [11.724 ns 11.763 ns]
  >    R^2  0.0296366 0.0296345
> Estimating the statistics of the sample
  >   mean [11.738 ns 11.776 ns]
  > median [11.645 ns 11.678 ns]
  >    MAD [406.89 ps 445.48 ps]
  >     SD [746.95 ps 880.17 ps]
```